File: Python3/1029.py
_,*a = map(int,input().split())
_,*b = map(int,input().split())
mid = (len(a)+len(b)-1)>>1
a.append(1000000000) #防止越界
b.append(1000000000)
i = 0
j = 0
index = 0
while index < mid:
    if a[i] < b[j]:
        i += 1
    else:
        j += 1
    index += 1
if a[i]<b[j]:
    print(a[i])
else:
    print(b[j])

File: c++/1029.cpp
//
//  main.cpp
//  test
//
//  Created by 李寻欢 on 2019/10/1.
//  Copyright © 2019 李寻欢. All rights reserved.
//
#include <stdio.h>
#include <iostream>
#include <string>
#include <map>
#include <cmath>
#include <algorithm>
#include <stack>
#include <vector>
#include <queue>
#include <unordered_map>
#include <string.h>
#include <set>
using namespace std;

const int MAXN = 2e5+10;
int n,m;
long long arr1[MAXN];
long long arr2[MAXN];

int main(){
    scanf("%d",&n);
    for(int i=0;i<n;++i) scanf("%lld",&arr1[i]);
    scanf("%d",&m);
    for(int i=0;i<m;++i) scanf("%lld",&arr2[i]);
    arr1[n]=arr2[m] = 0x7fffffff;
    int mid = (n+m-1)>>1;
    int index = 0;
    int i=0,j=0;
    while (index <mid) {
        if(arr1[i]<arr2[j]) i++;
        else j++;
        index++;
    }
    if(arr1[i] <arr2[j]) printf("%lld",arr1[i]);
    else printf("%lld",arr2[j]);
}

